.github/workflows/maven.yml

```
@@ -55,14 +55,34 @@ jobs:
       with:
         name: liquibase-changelogs
         path: waltz-data/src/main/ddl/liquibase/
-    
+
     - name: Publish Test Report
       if: ${{ always() }}
       uses: scacap/action-surefire-report@v1
 
+    - name: Login to GitHub Registry
+      if: ${{ github.ref_type == 'tag' }}
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build and push Docker image
+      if: ${{ github.ref_type == 'tag' }}
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        push: ${{ github.ref_type == 'tag' }}
+        tags: |
+          ghcr.io/${{ github.repository }}:postgresql-${{ github.ref_name }}
+          ghcr.io/${{ github.repository }}:${{ github.ref_name }}
+          ghcr.io/${{ github.repository }}:postgresql 
+          ghcr.io/${{ github.repository }}:latest
```

nitpick: I _think_ we should use `postgres` instead of `postgresql` to be consistent with other uses (maven profile, image name, default host, etc.)
```suggestion
          ghcr.io/${{ github.repository }}:postgres-${{ github.ref_name }}
          ghcr.io/${{ github.repository }}:${{ github.ref_name }}
          ghcr.io/${{ github.repository }}:postgres
          ghcr.io/${{ github.repository }}:latest
```